config/lib/seed.js

```diff
@@ -2,93 +2,145 @@
 
 var mongoose = require('mongoose'),
   chalk = require('chalk'),
-  crypto = require('crypto'),
-  User = mongoose.model('User');
+  crypto = require('crypto');
 
-console.log(chalk.bold.red('Warning:  Database seeding is turned on'));
-
-var seedUser = {
-  username: 'user',
-  password: 'User_Password1!',
-  provider: 'local',
-  email: 'user@localhost.com',
-  firstName: 'User',
-  lastName: 'Local',
-  displayName: 'User Local',
-  roles: ['user']
-};
-
-var seedAdmin = {
-  username: 'admin',
-  password: 'Admin_Password1!',
-  provider: 'local',
-  email: 'admin@localhost.com',
-  firstName: 'Admin',
-  lastName: 'Local',
-  displayName: 'Admin Local',
-  roles: ['user', 'admin']
-};
-
-//If production only seed admin if it does not exist
-if (process.env.NODE_ENV === 'production') {
-  //Add Local Admin
-  User.find({username: seedAdmin.username}, function (err, users) {
-    if (users.length === 0) {
-      var user = new User(seedAdmin);
-
-      // generate a random password and save 
-      User.generateRandomPassphrase()
-      .then(saveUser(user))
-      .catch(reportError);
-
-    } else {
-      console.log(seedAdmin.username + ' user exists');
-    }
+function removeUser (user) {
+  return new Promise(function (resolve, reject) {
+    var User = mongoose.model('User');
+    User.find({username: user.username}).remove(function (err) {
+        if (err) {
+          reject(new Error('Database Seeding:\t\t\tFailed to remove local ' + user.username));
+        }
+        resolve();
+    });
   });
-} else {
-
-  //Add Local User
-  User.find({username: seedUser.username}).remove(function () {
-    var user = new User(seedUser);
+}
 
-    // generate a random password and save
-    User.generateRandomPassphrase()
-    .then(saveUser(user))
-    .catch(reportError);
-  });
+function saveUser (user) {
+  return function() {
```

Comment:
We need the closure there so that saveUser can act as a promise, and thus it will call the resolve() with the data variable of `theuser`.
Try commenting out the closure function and test to see what I mean.